c++/src/capnp/compiler/generics.c++

```diff
@@ -131,10 +131,19 @@ bool BrandedDecl::compileAsType(
         }
 
         if (elementType.isAnyPointer()) {
-          addError(errorReporter, "'List(AnyPointer)' is not supported.");
-          // Seeing List(AnyPointer) later can mess things up, so change the type to Void.
-          elementType.setVoid();
-          return false;
+          auto unconstrained = elementType.getAnyPointer().getUnconstrained();
+
+          if (unconstrained.isAnyKind()) {
+            addError(errorReporter, "'List(AnyKind)' is not supported.");
```

Comment:
Let's change the error message back to say `List(AnyPointer)`, since that's what someone would actually type in their code to trigger this error. There is no type called `AnyKind`.